lib/src/main/kotlin/StatefulSolver.kt

```diff
@@ -118,143 +95,42 @@ class StatefulSolver<TState, TAction>(
             }
 
             currentState = newState
-            depth += 2
+            depth++
```

Comment:
Note to @JunTaoLuo , explain why its increment by 1 and not 2 now.

Reply:
It used to be two layers of nodes (ActionNode and StateNode) but it didn't really make sense how we were counting layers differently. Looking at other implementations and papers, it makes more sense to use one layer of nodes but have more complexity for how action/state relationships are maintained in the node.